decode.go

```diff
@@ -400,8 +400,17 @@ func (opts DecOptions) decMode() (*decMode, error) {
 
 // DecMode is the main interface for CBOR decoding.
 type DecMode interface {
+	// Unmarshal parses the CBOR-encoded data into the value pointed to by v
+	// using the decoding mode.  If v is nil, not a pointer, or a nil pointer,
+	// Unmarshal returns an error.
+	//
+	// See the documentation for Unmarshal for details.
 	Unmarshal(data []byte, v interface{}) error
+	// Valid checks whether the CBOR data is complete and well-formed.
+	Valid(data []byte) error
+	// NewDecoder returns a new decoder that reads from r using dm DecMode.
 	NewDecoder(r io.Reader) *Decoder
+	// DecOptions returns user specified options used to create this DecMode.
```

Comment:
Thanks for adding comments to existing functions :+1: 

Maybe we can add another `Valid` function at package level for default decoding mode.  This allows users of default and custom decoding modes to have the same API.

```
// Valid checks whether the CBOR data is complete and well-formed.
func Valid(data []byte) error {
	return defaultDecMode.Valid(data)
}
```

Reply:
Great idea, added in 25cb6233b1240c374944e6f9d4a36fb358398420